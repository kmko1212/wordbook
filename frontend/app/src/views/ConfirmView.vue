<script setup>
import { ref, reactive, onMounted } from 'vue'
import { useRouter } from 'vue-router'

const router = useRouter()
const word = ref('')
const meaning = ref('')
const key = ref(1)
const wordObj = reactive({
  word: '',
  meaning: ''
})

onMounted(() => {
  getWord(key.value)
})

const toNextWord = () => {
  key.value++
  if (!localStorage.getItem(key.value)) key.value = 1
  getWord(key.value)
}

const getWord = (wordKey) => {
  Object.assign(wordObj, JSON.parse(localStorage.getItem(wordKey)))
  word.value = wordObj.word
  meaning.value = wordObj.meaning
}

const toHomeView = async () => {
  try {
    await router.push('/')
  } catch (error) {
    console.error(error)
  }
}
</script>

<template>
  <v-container>
    <v-sheet class="w-50">
      <div class="text-h3 pb-5">確認</div>
      <p class="text-subtitle-1">英単語:</p>
      <p class="text-h5">{{ word }}</p>
      <p class="text-subtitle-1">意味:</p>
      <p class="text-h5">{{ meaning }}</p>
      <v-btn class="ml-5 mt-5" id="button-next" @click="toNextWord">次へ</v-btn>
      <v-btn class="ml-5 mt-5" id="button-home" @click="toHomeView">Homeへ戻る</v-btn>
    </v-sheet>
  </v-container>
</template>
