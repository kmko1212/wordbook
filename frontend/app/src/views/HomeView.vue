<script setup>
import { useRouter } from 'vue-router'
const router = useRouter()
const toResisterView = async () => {
  try {
    await router.push('/resister')
  } catch (error) {
    console.error(error)
  }
}
const toConfirmView = async () => {
  try {
    await router.push('/confirm')
  } catch (error) {
    console.error(error)
  }
}
</script>

<template>
  <v-container>
    <v-btn
      color="primary"
      size="x-large"
      rounded="lg"
      variant="flat"
      class="ml-5 mt-5"
      id="button-resister"
      @click="toResisterView"
      >登録</v-btn
    >
    <v-btn
      color="secondary"
      size="x-large"
      rounded="lg"
      variant="flat"
      class="ml-5 mt-5"
      id="button-confirm"
      @click="toConfirmView"
      >確認</v-btn
    >
  </v-container>
</template>
