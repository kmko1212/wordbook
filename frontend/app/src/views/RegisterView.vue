<script setup>
import { ref } from 'vue'
import { useRouter } from 'vue-router'

const router = useRouter()
const word = ref('')
const meaning = ref('')
const key = ref(0)

const onResister = () => {
  setKey()
  console.log(key.value)
  localStorage.setItem(
    key.value,
    JSON.stringify({
      word: word.value,
      meaning: meaning.value
    })
  )
  localStorage.setItem('wordCount', key.value)
}

const setKey = () => {
  if (!localStorage.getItem('wordCount')) {
    localStorage.setItem('wordCount', 0)
  } else {
    key.value = Number(localStorage.getItem('wordCount')) + 1
  }
}

const toHomeView = async () => {
  try {
    await router.push('/')
  } catch (error) {
    console.error(error)
  }
}
</script>

<template>
  <v-container>
    <v-sheet class="w-50">
      <div class="text-h3 pb-5">登録</div>
      <v-text-field
        label="英単語"
        placeholder="英単語を入力"
        type="input"
        v-model="word"
      ></v-text-field>
      <v-text-field
        label="意味"
        placeholder="意味を入力"
        type="input"
        v-model="meaning"
      ></v-text-field>
      <v-btn class="ml-5 mt-5" id="button-register" @click="onResister">登録</v-btn>
      <v-btn class="ml-5 mt-5" id="button-home" @click="toHomeView">Homeへ戻る</v-btn>
    </v-sheet>
  </v-container>
</template>
