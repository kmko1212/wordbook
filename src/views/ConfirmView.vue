<script setup>
import { ref, reactive, onMounted } from 'vue'
import { useRouter } from 'vue-router'

const router = useRouter()
const word = ref('')
const meaning = ref('')
const key = ref(1)
const wordObj = reactive({
  word: '',
  meaning: ''
})

onMounted(() => {
  getWord(key.value)
})

const toNextWord = () => {
  key.value++
  if (!localStorage.getItem(key.value)) key.value = 1
  getWord(key.value)
}

const getWord = (wordKey) => {
  Object.assign(wordObj, JSON.parse(localStorage.getItem(wordKey)))
  word.value = wordObj.word
  meaning.value = wordObj.meaning
}

const toHomeView = async () => {
  try {
    await router.push('/')
  } catch (error) {
    console.error(error)
  }
}
</script>

<template>
  <div>
    <h2>確認</h2>
    <p>英単語: {{ word }}</p>
    <p>意味: {{ meaning }}</p>
    <p><button id="button-next" @click="toNextWord">次へ</button></p>
    <p><button id="button-home" @click="toHomeView">Homeへ戻る</button></p>
  </div>
</template>
