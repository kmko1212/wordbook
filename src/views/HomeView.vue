<script setup>
import { useRouter } from 'vue-router'
const router = useRouter()
const toResisterView = () => {
  router.push('/resister')
}
const toConfirmView = () => {
  router.push('/confirm')
}
</script>

<template>
  <div>
    <h2>Wordbook</h2>
    <button @click="toResisterView">登録</button>
    <button @click="toConfirmView">確認</button>
  </div>
</template>
