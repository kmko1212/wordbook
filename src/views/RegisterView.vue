<script setup>
import { ref } from 'vue'
import { useRouter } from 'vue-router'

const router = useRouter()
const word = ref('')
const meaning = ref('')
const key = ref(0)

const onResister = () => {
  setKey()
  localStorage.setItem(
    key.value,
    JSON.stringify({
      word: word.value,
      meaning: meaning.value
    })
  )
  localStorage.setItem('wordCount', key.value)
}
const setKey = () => {
  if (!localStorage.getItem('wordCount')) {
    localStorage.setItem('wordCount', 0)
  } else {
    key.value = Number(localStorage.getItem('wordCount')) + 1
  }
  console.log('setKey')
}
const toHomeView = () => {
  router.push('/')
}
</script>

<template>
  <div>
    <h2>登録</h2>
    <p>英単語: {{ word }}</p>
    <input v-model="word" placeholder="英単語を入力" />
    <p>意味: {{ meaning }}</p>
    <input v-model="meaning" placeholder="意味を入力" />
    <p><button @click="onResister">登録</button></p>
    <p><button @click="toHomeView">Homeへ戻る</button></p>
  </div>
</template>
