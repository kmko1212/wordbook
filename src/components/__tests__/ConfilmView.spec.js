import { createRouter, createMemoryHistory } from 'vue-router'
import { describe, it, expect, beforeEach } from 'vitest'
import { reactive } from 'vue'

import { mount } from '@vue/test-utils'
import HomeView from '../../views/HomeView.vue'
import RegisterView from '../../views/RegisterView.vue'
import ConfirmView from '../../views/ConfirmView.vue'

const wordObj = reactive({
  word: 'apple',
  meaning: 'りんご'
})

let router

beforeEach(() => {
  router = createRouter({
    history: createMemoryHistory(),
    routes: [
      {
        path: '/',
        name: 'home',
        component: HomeView
      },
      {
        path: '/resister',
        name: 'resister',
        component: RegisterView
      },
      {
        path: '/confirm',
        name: 'confirm',
        component: ConfirmView
      }
    ]
  })
  localStorage.setItem('wordCount', 3)
  localStorage.setItem(
    1,
    JSON.stringify({
      word: 'apple',
      meaning: 'りんご'
    })
  )
  localStorage.setItem(
    2,
    JSON.stringify({
      word: 'orange',
      meaning: 'みかん'
    })
  )
  localStorage.setItem(
    3,
    JSON.stringify({
      word: 'banana',
      meaning: 'バナナ'
    })
  )
})

describe('ConfilmView', () => {
  it('renders properly', async () => {
    const wrapper = mount(ConfirmView, { global: { plugins: [router] } })
    const vm = wrapper.vm
    vm.word = wordObj.word
    vm.meaning = wordObj.meaning
    await wrapper.vm.$nextTick()
    expect(wrapper.html()).toMatchSnapshot()
  })
  it('navigates to / when #button-home is clicked', async () => {
    const wrapper = mount(ConfirmView, { global: { plugins: [router] } })

    await wrapper.find('#button-home').trigger('click')

    await router.isReady()

    expect(router.currentRoute.value.path).toBe('/')
  })
  it('when getWord is called with 3', () => {
    const wrapper = mount(ConfirmView, { global: { plugins: [router] } })
    wrapper.vm.getWord(3)
    expect(wrapper.vm.word).toBe('banana')
    expect(wrapper.vm.meaning).toBe('バナナ')
  })
  it('when toNextWord is called', async () => {
    const wrapper = mount(ConfirmView, { global: { plugins: [router] } })

    const vm = wrapper.vm
    vm.key = 1
    vm.word = 'りんご'
    vm.meaning = 'apple'
    await wrapper.vm.$nextTick()

    await wrapper.find('#button-next').trigger('click')
    expect(wrapper.vm.word).toBe('orange')
    expect(wrapper.vm.meaning).toBe('みかん')
  })
})
