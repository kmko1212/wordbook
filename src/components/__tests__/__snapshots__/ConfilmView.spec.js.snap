// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`ConfilmView > renders properly 1`] = `
"<div>
  <h2>確認</h2>
  <p>英単語: apple</p>
  <p>意味: りんご</p>
  <p><button>次へ</button></p>
  <p><button>Homeへ戻る</button></p>
</div>"
`;
